### PR TITLE
difference-of-squares: renamed occurances of sums to sum

### PR DESCRIPTION
**Pull request by Scientifica96**
Renamed occurrences of sums to sum, like in `square_of_sums`.

### CODE REVIEW

File: exercises/difference-of-squares/difference_of_squares.exs [hunk 1]
```diff
@@ -1,6 +1,6 @@
 defmodule Squares do
   @moduledoc """
-  Calculate sum of squares, square of sums, difference between two sums from 1 to a given end number.
+  Calculate sum of squares, square of sum, difference between two sum from 1 to a given end number.
```
Comment (devonestes) [inline exercises/difference-of-squares/difference_of_squares.exs]: This should still be `difference between two sums from...`

File: exercises/difference-of-squares/example.exs [hunk 2]
```diff
@@ -1,6 +1,6 @@
 defmodule Squares do
   @moduledoc """
-  Calculate sum of squares, square of sums, difference between two sums from 1 to a given end number.
+  Calculate sum of squares, square of sum, difference between two sum from 1 to a given end number.
```
Comment (devonestes) [inline exercises/difference-of-squares/example.exs]: This should still be `difference between two sums from...`

File: exercises/difference-of-squares/difference_of_squares.exs (hunk 1 above)
Comment (Scientifica96) [reply exercises/difference-of-squares/difference_of_squares.exs]: Thank you for your remark. I'll change it now!

File: exercises/difference-of-squares/example.exs (hunk 2 above)
Comment (Scientifica96) [reply exercises/difference-of-squares/example.exs]: Thank you for your remark. I'll change it now.

**Pull request merged**